### PR TITLE
T140217 Adjust texvcjs to prevent whitespace modifications in ce-tags

### PR DESCRIPTION
**Pull request by manfredschaefer**
Implemented grammar of `mhchem` in the `texvcjs` parser.
To switch to math mode within a chemical formula `\begin{math} some math \end{math}` is used as `$` is not a option (security reasons). Currently only named colors are supported within `\color{}{}` (without a backslash).

### CODE REVIEW

**Comment by physikerwelt** [comment]
@manfredschaefer can you remove 3d2a8d3 from the PR

File: lib/ast.js [hunk 1]
```diff
@@ -142,6 +142,7 @@ var RenderT = module.exports.RenderT = new Enum( 'RenderT', {
 var Tex = module.exports.Tex = new Enum( 'Tex', {
     LITERAL: { args: [ RenderT ] }, // contents
     CURLY:   { args: [ ['self'] ] }, // expr
+    DOLLAR:  { args: [ ['self'] ] }, // expr
```
Comment (physikerwelt) [inline lib/ast.js]: Why is this needed?

File: lib/astutil.js [hunk 2]
```diff
@@ -134,6 +147,10 @@ ast.Tex.defineVisitor("contains_func", {
         // { tl1 tl2 tl3 ... }
         return arr_contains_func(tl, target);
     },
+    DOLLAR: function(target, tl) {
```
Comment (physikerwelt) [inline lib/astutil.js]: why?

File: lib/ast.js (hunk 1 above)
Comment (manfredschaefer) [inline lib/ast.js]: I use this for rewriting `\begin{math}...\end{math}` to `$...$` as dollar signs cannot be used in the input and mhchem does not any other way to switch to math mode (that I could find) than the dollar sign

**Comment by physikerwelt** [comment]
wait for
physikerwelt/mathpipe#16
before merge

**Comment by physikerwelt** [comment]
@d00rman I think we should merge this and release it as 0.3.3-alpha.1 and use that alpha version for our regression test. Are you OK with this. Or will there by automatic deployments from the master branch of this repo?

**Comment by physikerwelt** [comment]
I added more changes here #24 so I close this one
